jsonschema/validators.py

```diff
@@ -161,15 +161,22 @@ def iter_errors(self, instance, _schema=None):
             if scope:
                 self.resolver.push_scope(scope)
             try:
-                validators = []
-
-                # We make sure $ref is evaluated first if available in schema
-                ref = _schema.get(u"$ref")
-                if ref is not None:
-                    validators = [(u"$ref", ref)]
-
-                # Add all remaining validators
-                validators += [(x, _schema[x]) for x in _schema if x not in ["$ref"]]
+                if isinstance(self, Draft202012Validator):
+                    validators = []
+                    # We make sure $ref is evaluated first if available in schema
+                    ref = _schema.get(u"$ref")
+                    if ref is not None:
+                        validators = [(u"$ref", ref)]
```

Comment:
These 3 lines could be factored out of the if/else.

Reply:
Done